From cf1843aca8c0c83868325a32b59f3202e9bed70c Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Thu, 7 Apr 2016 16:25:21 +0300
Subject: [PATCH 374/666] powerpc/85xx: drop generic tb sync

These platforms rely on a guts based tb sync
mechanism so no need to default to the generic
tb sync.
This also fixes a kvm over-subscription scenario
where several vcpus of a booting kvm guest share
the same cpu. In such a scenario, the generic
implementation of give_timebase() takes minutes
to finish (~3mins seen on a p5020ds).

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/powerpc/platforms/85xx/smp.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/powerpc/platforms/85xx/smp.c b/arch/powerpc/platforms/85xx/smp.c
index f51fd35..c0481ca 100644
--- a/arch/powerpc/platforms/85xx/smp.c
+++ b/arch/powerpc/platforms/85xx/smp.c
@@ -351,10 +351,6 @@ struct smp_ops_t smp_85xx_ops = {
 	.cpu_disable	= generic_cpu_disable,
 	.cpu_die	= generic_cpu_die,
 #endif
-#if defined(CONFIG_KEXEC_CORE) && !defined(CONFIG_PPC64)
-	.give_timebase	= smp_generic_give_timebase,
-	.take_timebase	= smp_generic_take_timebase,
-#endif
 };
 
 #ifdef CONFIG_KEXEC_CORE
-- 
2.7.4

